authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -278,8 +278,9 @@ private void removeSchemaMetadataObject(AuthorizationMetadataObject authzMetadat
         authzMetadataObject instanceof PathBasedMetadataObject,
         "The metadata object must be a PathBasedMetadataObject");
     Preconditions.checkArgument(
-        authzMetadataObject.type()
-            == PathBasedMetadataObject.PathType.get(MetadataObject.Type.SCHEMA),
+        authzMetadataObject
+            .type()
+            .equals(PathBasedMetadataObject.PathType.get(MetadataObject.Type.SCHEMA)),
```

Comment:
Why don't you create some static instances in PathBasedMetadataObject to avoid creating new instances every time?

```java
public class PathBasedMetadataObject {

public static final SCHEMA_PATH = new PathType(MetadataObject.Type.SCHEMA);
public static final TABLE_PATH = new PathType(MetadataObject.Type.TABLE);
public static final FILESET_PATH = new PathType(MetadataObject.Type.FILESET);
...
}
```

Then we can use `SCHEMA_PATH`, `TABLE_PATH` and `FILESET_PATH` directly.